PEP 670 amendments (#2157)
python/peps: pep-0670.rst

* Compatibility layer for different C compilers, C language extensions,
  or recent C features.
  Example: ``#define Py_ALWAYS_INLINE __attribute__((always_inline))``.

> * Macros that need the stringification features of the C preprocessor.  Such
>   macros should be matched with an ``#undef`` after use, if possible.
> * Macros that need the concatenation features of the C preprocessor.  Such
>   macros should be matched with an ``#undef`` after use, if possible.

Reviewer: ```suggestion
* Macros that need the stringification or concatenation feature of the C preprocessor.
```

internal C API exposes implementation details by design and should not be
used outside Python.

The performance impact must be measured and regarded before converting a macro

> to a static inline function or a regular function.

Reviewer: The first paragraph of this section is not enough?

I'm not aware of any performance slowdown of converting a macro to a static inline function. I wouldn't require to run a benchmark for this case.

Author: I guess the first paragraph is enough. There is no need in repeating stuff already said.
